src/actions/restaurantsActions.js

```diff
@@ -0,0 +1,35 @@
+export function fetchRestaurants(url) {
+    return function(dispatch){
+        fetch(url)
+            .then(res => res.json())
+            .then(
+                (result) => {
+                    dispatch({type: "FETCH_RESTAURANTS_FULFILLED", payload: result})
+                },
+                (error) => {
+                    dispatch({type: "FETCH_RESTAURANTS_REJECTED", payload: error})
+                }
+            )
+    }
+}
```

Comment:
Cool.
Got a few suggestions:
1. It's a common practice to extract the part that creates the action (or _action creator_). Basically into a function like:
   ```js
   const fetchRestaurantSuccess = (payload) => ({ type: ..., payload });
   ```
1. It's also common practice to extract the type into a constant so it can be reused in the reducer.
   ```js
   const FETCH_RESTAURANTS_FULFILLED = 'FETCH_RESTAURANTS_FULFILLED';
  ```
1. I personally like redux-actions for this: https://redux-actions.js.org/api/createaction#createactiontype
   ```
   const fetchRestaurantSuccess = createAction(FETCH_RESTAURANTS_FULFILLED);
   ```
1. And lastly, I recommend redux-action's `handleActions` for creating the reducer: https://redux-actions.js.org/api/handleaction#handleactionsreducermap-defaultstate